canonicalwebteam/store_api/store.py

```diff
@@ -102,64 +27,119 @@ def process_response(self, response):
                     if "error_list" in body
                     else body["error-list"]
                 )
-                api_error_exception = ApiResponseErrorList(
-                    "The api returned a list of errors",
+                api_error_exception = StoreApiResponseErrorList(
+                    "The API returned a list of errors",
                     response.status_code,
                     error_body,
                 )
                 raise api_error_exception
             else:
-                raise ApiResponseError(
-                    "Unknown error from api", response.status_code
+                raise StoreApiResponseError(
+                    "Unknown error from API", response.status_code
                 )
 
         return body
 
-    def get_all_snaps(self, size):
-        all_snaps_response = self.session.get(ALL_SNAPS_URL.format(size=size))
-
-        return self.process_response(all_snaps_response)
-
-    def get_featured_snaps(self):
-        featured_response = self.session.get(FEATURE_SNAPS_URL)
+    def get_endpoint_url(self, endpoint, api_version=1):
+        url = [self.config[api_version]["base_url"], endpoint]
 
-        return self.process_response(featured_response)
+        return "".join(url)
 
-    def get_promoted_snaps(self):
-        promoted_response = self.session.get(PROMOTED_QUERY_URL)
-
-        return self.process_response(promoted_response)
+    def get_all(self, size, api_version=1):
+        url = self.get_endpoint_url(
+            "search", {"scope": "wide", "size": size}, api_version
+        )
 
-    def get_searched_snaps(self, snap_searched, size, page, category=None):
-        url = SNAP_SEARCH_URL.format(
-            snap_name=snap_searched, size=size, page=page
+        return self.process_response(
+            self.session.get(
+                url, headers=self.config[api_version].get("headers")
+            )
         )
 
+    def get_searcheds(self, search, size, page, category=None, api_version=1):
```

Comment:
Weird function name, typo?

Reply:
Do you think I should change it to `get_seach`? Not exactly a typo it's because I removed the word snap from there